targeteditor/src/main/scala/explore/targeteditor/ElevationPlot.scala

```diff
@@ -0,0 +1,142 @@
+) extends ReactProps[ElevationPlot](ElevationPlot.component)
+
+object ElevationPlot {
+  type Props = ElevationPlot
+
+  implicit val propsReuse: Reusability[Props] = Reusability.derive
+
+  val PlotPeriod: Duration = Duration.ofHours(12)
+  val PlotEvery: Duration  = Duration.ofMinutes(1)
+
+  class Backend( /*$ : BackendScope[Props, State]*/ ) {
+    def render(props: Props) = {
+
+      val start      = Instant.now()
+      val end        = start.plus(PlotPeriod)
+      val elevations =
+        ElevationCalc.elevationsForInterval(props.site, start, end, PlotEvery, _ => props.coords)
+
+      val options = Options()
+        .setChart(ChartOptions().setHeight(280))
+        .setTitle(TitleOptions().setText(""))
+        .setLegend(LegendOptions().setEnabled(false))
+        .setCredits(CreditsOptions().setEnabled(false))
+        .setTooltip(
+          TooltipOptions()
+            .setXDateFormat("%H:%M")
+            .setHeaderFormat("<b>{point.key}</b>")
+            .setPointFormat("{point.y}")
+            .setPointFormatter { (p: Point) =>
+              p.y.toOption
+                .map(d => s"<br/>${Angle.fromStringDMS.reverseGet(Angle.fromDoubleDegrees(d))}")
+                .orEmpty
+            }
+            .setSnap(0)
+            .setAnimation(false)
+            .setShape(TooltipShapeValue.square)
+        )
+        .setXAxis(XAxisOptions().setType(AxisTypeValue.datetime))
+        .setYAxis(
+          List(
+            YAxisOptions()
+              .setTitle(YAxisTitleOptions().setText(""))
+              .setAllowDecimals(false)
+              .setMin(0)
+              .setMax(90)
+              .setTickInterval(10)
+              .setLabels(YAxisLabelsOptions().setFormat("{value}°")),
+            YAxisOptions()
+              .setLinkedTo(0)
+              .setOpposite(true)
+              .setTitle(YAxisTitleOptions().setText(""))
+              .setLabels(YAxisLabelsOptions().setFormat("{value}°"))
+          ).toJSArray
+        )
+        .setPlotOptions(
+          PlotOptions()
+            .setSeries(
+              PlotSeriesOptions()
+                .setLineWidth(4)
+                .setMarker(PointMarkerOptionsObject().setEnabled(false))
+                .setStates(
+                  SeriesStatesOptionsObject()
+                    .setHover(SeriesStatesHoverOptionsObject().setEnabled(false))
+                )
+            )
+        )
+        .setSeries(
+          List(
+            SeriesLineOptions(line)
+              .setName("Elevation")
+              .setData(
+                elevations.collect {
+                  case (instant, angle) if angle.toSignedDoubleDegrees <= 90 =>
+                    PointOptionsObject()
+                      .setX(instant.toEpochMilli.toDouble)
+                      .setY(angle.toSignedDoubleDegrees)
+                      .asInstanceOf[Chart.Data]
+                }.toJSArray
+              )
+          ).map(_.asInstanceOf[SeriesOptionsType]).toJSArray
```

Comment:
Maybe we can use `collect` here?

Reply:
With the current typings generated by ScalablyTyped, I don't see a way to avoid the casting. This may even be an issue in the TS declarations.

`setSeries` is defined as `setSeries(value: js.Array[SeriesOptionsType])` and `SeriesOptionsType` is unrelated at all to `SeriesLineOptions` (or options for series of other types). Maybe we can hide the castings in the facade.